### ISSUE TITLE
The "smart" mode cannot detect the right modified files

### ISSUE THREAD

**Issue by haozhestat** [Azure/shrike#130]
**Before filing, make sure to upgrade to the latest shrike version and check if the bug reproduces.**

**Category**
Please select:
- [ ] shrike.pipeline
- [x] shrike.build
- [ ] shrike.compliant_logging

**Describe the bug**
This is a bug reported by Zhi Chen (Search Relevance Team):
He updated module: pyspark_related_search_aig_session in PR: [Pull request 23379: change module for new version](https://eemo.visualstudio.com/TEE/_git/TEEGit/pullrequest/23379?path=%2FOffline%2FSearchRelevance%2Fpython%2Fsrc%2Fsteps%2Fpyspark-related-search-aig-session%2Frun.py&_a=files)
Then this PR triggered the sign build of pipeline: [Pipelines - Run 20210718.6](https://eemo.visualstudio.com/TEE/_build/results?buildId=872415&view=results)
From the log of above pipeline, it failed to detect his change to this module, but updated another unchanged module.

**Expected behavior**
The "smart" mechanism should detect the module "pyspark-related-search-aig-session" rather than "related-search-index-gen".

**Version info**
1.7.2

**Comment by thomasp-ms** [comment]
**TLDR: transient issues with the repo, I cannot reproduce, customer is unblocked.**

There were some issues with git around the time the build was triggered; [another build](https://eemo.visualstudio.com/TEE/_build/results?buildId=872452&view=results) failed at the "git checkout" phase at about the same time. I believe this is why the smart mode did not pick up the correct files: the git repo was in an unstable state.

Now the repo seems healthy again. I have checked ~6 subsequent builds and they all picked up the correct files.

I cannot reproduce this issue, so there's not much more I'm able to do. I see the customer did the right thing: change the module version to trigger [another build](https://eemo.visualstudio.com/TEE/_build/results?buildId=872543&view=results), which did register the module properly.
